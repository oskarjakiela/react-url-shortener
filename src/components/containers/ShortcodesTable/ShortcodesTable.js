import { map } from 'ramda';
import React from 'react';
import timeago from 'timeago.js';

import './ShortcodesTable.css';
import { Shortcode } from 'components/containers';
import { Table, Text } from 'components/ui';


const displayName = 'Shorty-ShortcodesTable';

const headers = ['Link', 'Visits', 'Last Visited'];

const mapItemsToRows = (items) => map(({
  lastSeenDate,
  redirectCount,
  shortcode,
  url,
}) => {
  return [
    <Shortcode
      value={shortcode}
      url={url}
    />,
    <Text>{redirectCount}</Text>,
    <Text>{timeago().format(lastSeenDate)}</Text>,
  ];
}, items);

const ShortcodesTable = ({ items }) => (
  <div className={displayName}>
    <Table
      headers={headers}
      rows={mapItemsToRows(items)}
    />
  </div>
);

ShortcodesTable.displayName = displayName;

export default ShortcodesTable;
