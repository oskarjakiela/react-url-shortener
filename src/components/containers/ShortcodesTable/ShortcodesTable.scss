 .Shorty-ShortcodesTable {
  .Shorty-Table {
    td,
    th {
      &:first-child {
        text-align: left;
      }
    }
  }

  .Shorty-Shortcode {
    max-width: 26rem;
  }
}
