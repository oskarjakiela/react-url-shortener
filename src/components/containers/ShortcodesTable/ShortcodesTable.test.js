import React from 'react';
import renderer from 'react-test-renderer';

import ShortcodesTable from './ShortcodesTable';


it('<ShortcodesTable /> renders correctly', () => {
  const tree = renderer
    .create(<ShortcodesTable
      items={[{
        lastSeenDate: '2018-02-02T18:25:43.511Z',
        redirectCount: 1140,
        shortcode: 'fca1ec3fe',
        startDate: '2018-02-01T11:49:02.283Z',
        url: 'https://jlongster.com/Radical-Statements-about-the-Mobile-Web',
      }]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
