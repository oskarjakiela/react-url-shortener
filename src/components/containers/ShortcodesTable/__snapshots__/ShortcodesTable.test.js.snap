// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortcodesTable /> renders correctly 1`] = `
<div
  className="Shorty-ShortcodesTable"
>
  <table
    className="Shorty-Table"
  >
    <thead>
      <tr>
        <th>
          Link
        </th>
        <th>
          Visits
        </th>
        <th>
          Last Visited
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="Shorty-Shortcode"
          >
            <span
              className="Shorty-Shortcode__url-body"
            >
              <span
                className="Shorty-Shortcode__domain"
              >
                shooooort.com/
              </span>
              <span
                className="Shorty-Shortcode__value"
              >
                fca1ec3fe
              </span>
            </span>
            <a
              className="Shorty-Link Shorty-Shortcode__copy-link"
            >
              Click to copy this link
            </a>
            <br />
            <span
              className="Shorty-Text Shorty-Text--unimportant Shorty-Shortcode__url"
            >
              https://jlongster.com/Radical-Statements-about-the-Mobile-Web
            </span>
          </div>
        </td>
        <td>
          <span
            className="Shorty-Text Shorty-Text--primary"
          >
            1140
          </span>
        </td>
        <td>
          <span
            className="Shorty-Text Shorty-Text--primary"
          >
            2 days ago
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
